lib/std/event/lock.zig

```diff
@@ -16,107 +16,90 @@ const Loop = std.event.Loop;
 /// Allows only one actor to hold the lock.
 /// TODO: make this API also work in blocking I/O mode.
 pub const Lock = struct {
-    shared: bool,
-    queue: Queue,
-    queue_empty: bool,
+    mutex: std.Mutex = std.Mutex{},
+    head: usize = UNLOCKED,
```

Comment:
consider a type: 
```zig
enum(usize) {
  UNLOCKED = 0,
  LOCKED = 0x45,

  pub fn toPointer(self: @This()) *Waiter {
    return @intToPtr(*Waiter, @enumToInt(self));
  }

  _
}`

Reply:
Added documentation on the `usize`'s use and believe its a good compromise over converting it to an enum